daml-assistant/daml-helper/src/DA/Daml/Helper/Studio.hs

```diff
@@ -156,19 +158,31 @@ getInstalledExtensions = do
     oldBundled <- getOldExt
     extensions <- getExtensions
     let oldBundledIsInstalled = isJust oldBundled
-        publishedExtensionIsInstalled = not oldBundledIsInstalled && publishedExtensionName `elem` extensions
-        bundledInstalled = bundledExtensionName `elem` extensions
+        publishedExtensionIsInstalled =
+            not oldBundledIsInstalled &&
+                Lowercase publishedExtensionName `elem` extensions
+        bundledInstalled = Lowercase bundledExtensionName `elem` extensions
     pure InstalledExtensions {..}
     where getOldExt :: IO (Maybe FilePath)
           getOldExt = do
               extensionsDir <- getVsCodeExtensionsDir
               let oldBundledDir = extensionsDir </> oldBundledExtensionDirName
               exists <- doesPathExist oldBundledDir
               pure $ if exists then Just oldBundledDir else Nothing
-          getExtensions :: IO [String]
+
+          getExtensions :: IO [Lowercase]
           getExtensions = do
               (_exitCode, extensionsStr, _err) <- runVsCodeCommand ["--list-extensions"]
-              pure $ lines extensionsStr
+              pure $ map Lowercase $ lines extensionsStr
```

Comment:
I was thinking that and then concluded that I prefer to do it this way instead of having to remember to use lowercase everything before `elem`. This also captures an invariant which will be harder to miss for the next person making changes.

Reply:
The maybe call it `CaseInsensitive`?

In practice it won't matter given the number of elements involved but what lead me to make this remark initially was that the code is computing `map toLower` at worst twice on each element of `extension`: once when computing `publishedExtensionIsInstalled` and once when computing `bundledInstalled`. Again, given that we run this code once on small lists, this is bikeshedding.

Reply:
Oh, the PR appeared as "not merged" in my browser when I made this comment. But just after I submitted it, it turned purple. Then ignore my renaming suggestion, not worth another PR!